core/src/main/java/com/sequenceiq/cloudbreak/service/upgrade/ClusterUpgradeAvailabilityService.java

```diff
@@ -180,7 +185,18 @@ private Image getCurrentImageFromCatalog(String currentImageId, CloudbreakImageC
         return imageService.getImage(stack.getId());
     }
 
-    private CloudbreakImageCatalogV3 getImagesFromCatalog(String imageCatalogUrl) throws CloudbreakImageCatalogException {
+    private CloudbreakImageCatalogV3 getImagesFromCatalog(Workspace workspace,
+            com.sequenceiq.cloudbreak.cloud.model.Image image) throws CloudbreakImageCatalogException {
+        String imageCatalogName = image.getImageCatalogName();
+        String imageCatalogUrl;
+        try {
+            imageCatalogUrl = imageCatalogService.get(workspace.getId(), imageCatalogName).getImageCatalogUrl();
+            LOGGER.info("Image catalog with name {} and url {} is used for image filtering.", imageCatalogName, imageCatalogUrl);
+        } catch (NotFoundException ex) {
+            imageCatalogUrl = image.getImageCatalogUrl();
```

Comment:
if you declare `imageCatalogUrl` on line 191 with this as a default value, then you can omit this